server/bin/update-server-xml.py

```diff
@@ -57,27 +57,74 @@ def _add_attributes(self, node, attributes):
             logger.debug("Setting %s to %s on %s" % (k, v, node.name))
             node.setProp(k, v)
 
+    def _is_different_impl(self, cnode, nnode):
+        logger.debug("Comparing nodes: %s and %s" % (cnode.name, nnode.name))
+
+        # Check if the nodes have different names...
+        if cnode.name != nnode.name:
+            logger.debug("Node names differ: %s != %s" % (cnode.name, nnode.name))
+            return True
+
+        # Compare attributes immediately present on both nodes...
+        cattribs = {}
+        nattribs = {}
+
+        if cnode.properties:
+            for prop in cnode.properties:
```

Comment:
Use a dictionary comprehension.  It's more pythonic and more succinct.

```
cattribs = {n.name: n.content for n in cnode.properties if n.type == "attribute"}
```

Reply:
How does that handle the case of cnode.properties being non-iterable?

Reply:
Yeah, sorry, I left out that the `if cnode.properties` needs to remain.